### PR TITLE
Adding CDAP Stream Formatter Transform

### PR DESCRIPTION
**Pull request by nitinmotgi**
This PR can be reviewed and applied only after PR https://github.com/caskdata/hydrator-plugins/pull/36. This PR contains StreamWriter Transform. The transform will format the data in stream from the StructuredRecord. 
- [x] StreamFormatter Transform
- [x] StreamFormatter Test Case
- [x] Added entry into README.rst

### CODE REVIEW

File: transform-plugins/src/main/java/co/cask/hydrator/plugin/StreamFormatter.java [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+      Schema out = Schema.parseJson(config.schema);
+      List<Schema.Field> fields = out.getFields();
+
+      if (fields.size() != 2) {
+        throw new IllegalArgumentException("Output schema should have only two fields. One of type " +
+                                             "String for Stream body and other of type Map<String, String> for " +
+                                             "Stream header.");
+      }
+
+      // We have to make sure that output schema includes two fields and only two fields.
```
Comment (sreevatsanraman) [inline transform-plugins/src/main/java/co/cask/hydrator/plugin/StreamFormatter.java:97]: The check below will check if each field is either a map or a string.The check below doesn't catch if field[0] is String and field[1] is also a string. Same case with Map.

Comment (nitinmotgi) [inline transform-plugins/src/main/java/co/cask/hydrator/plugin/StreamFormatter.java:97]: Addressed comment @sreevatsanraman. Can you please review.

File: transform-plugins/src/test/java/co/cask/hydrator/plugin/StreamFormatterTest.java [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+                          .set("b", "2")
+                          .set("c", "3")
+                          .set("d", "4")
+                          .set("e", "5").build(), emitter);
+    Assert.assertEquals("1,2,3,4,5", emitter.getEmitted().get(0).get("body"));
+    Map<String, String> header = emitter.getEmitted().get(0).get("header");
+    Assert.assertEquals("2", header.get("b"));
+    Assert.assertEquals("3", header.get("c"));
+  }
+}
```
Comment (sreevatsanraman) [inline transform-plugins/src/test/java/co/cask/hydrator/plugin/StreamFormatterTest.java]: new line at the end.

File: transform-plugins/src/main/java/co/cask/hydrator/plugin/StreamFormatter.java [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+
+    public Config(String header, String body, String format, String schema) {
+      this.header = header;
+      this.body = body;
+      this.format = format;
+      this.schema = schema;
+    }
+  }
+}
+
```
Comment (sreevatsanraman) [inline transform-plugins/src/main/java/co/cask/hydrator/plugin/StreamFormatter.java]: nit: extra line. please remove.

**Comment by sreevatsanraman** [comment]
LGTM. Two minor comments. Fix and merge.

File: transform-plugins/src/test/java/co/cask/hydrator/plugin/StreamFormatterTest.java (hunk 2 above)
Comment (nitinmotgi) [inline transform-plugins/src/test/java/co/cask/hydrator/plugin/StreamFormatterTest.java]: done

**Comment by nitinmotgi** [comment]
addressed comments.

File: transform-plugins/src/main/java/co/cask/hydrator/plugin/StreamFormatter.java (hunk 3 above)
Comment (nitinmotgi) [inline transform-plugins/src/main/java/co/cask/hydrator/plugin/StreamFormatter.java]: done

**Pull request merged**